src/Compilers/CSharp/Portable/CSharpResources.resx

```diff
@@ -3614,6 +3620,9 @@ Give the compiler some way to differentiate the methods. For example, you can gi
   <data name="ERR_BadSpecialByRefLocal" xml:space="preserve">
     <value>Parameters or locals of type '{0}' cannot be declared in async methods or lambda expressions.</value>
   </data>
+  <data name="ERR_BadSpecialByRefIterator" xml:space="preserve">
+    <value>foreach statement cannot operate on variables of type '{0}' in async or iterator methods.</value>
```

Comment:
Actually "foreach statement cannot operate on ..." seems a common pattern among error messages, so we should either fix them all or none. 
I will enter a bug.

---
In reply to: [123128084](https://github.com/dotnet/roslyn/pull/20293#discussion_r123128084) [](ancestors = 123128084,122807111)